ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/KtLint.kt

```diff
@@ -490,6 +515,12 @@ object KtLint {
                 comment.replace(Regex("\\s"), " ").replace(" {2,}", " ").split(" ")
 
             private fun <T> List<T>.tail() = this.subList(1, this.size)
+
+            private val suppressAnnotationRuleMap = mapOf(
```

Comment:
I think this map could be moved to some config file, to ease addition of new suppression 
 rules

Reply:
I am not sure about having this mapping:
- "RemoveCurlyBracesFromTemplate" just becomes an alias to "string-template" rule
- This alias is detached from the rule itself
- Overall this PR goes into direction of adding another way for [suppress rules](https://github.com/pinterest/ktlint#how-do-i-suppress-an-error-for-a-lineblockfile)

To be consistent, I would propose to use following approach:
```kotlin
@Suppress("ktlint-string-template")
```
to disable rule for the block of code and remove this map.

Reply:
I agree that this map looks weird, however original purpose for this PR was to respect annotations like `@Suppress("RemoveCurlyBracesFromTemplate")` to disable *both* IntellijIDEA and ktlint checks at the same time. Otherwise we will have to add both annotations
```
@Suppress("ktlint-string-template")
@Suppress("RemoveCurlyBracesFromTemplate")
println("Not found: ${notFound}")
```

and this way all sense in introduction of support for `@Suppress` annotation seems lost to me, cause it becomes specific for ktlint, and as you mentioned we already have ways to suppress ktlint checks

Reply:
> I agree that this map looks weird, however original purpose for this PR was to respect annotations like @Suppress("RemoveCurlyBracesFromTemplate") to disable both IntellijIDEA and ktlint checks at the same time.

Sorry, missed that. Then, as intermediate step, having this map looks ok. In the future would be better that rule provides a list of suppress names aliases.